src/Shared/Interfaces/IDeviceClient.cs

```diff
@@ -13,7 +13,7 @@ public interface IDeviceClient
         Task SendMessageBatchAsync(IEnumerable<string> msgs);
         Task<Message> ReceiveMessageAsync();
         Task RegisterDirectMethodAsync(MethodCallback methodHandler);
-        Task RegisterDesiredPropertyUpdateCallbackAsync(DesiredPropertyUpdateCallback methodHandler);
+        void RegisterDesiredPropertyUpdateCallbackAsync(DesiredPropertyUpdateCallback methodHandler);
```

Comment:
Is this not async anymore ? can we remove the `Async` from the name ?

Reply:
while its no longer returning a task, the method implementation is still async.